src/app/content/components/ContentLink.spec.tsx

```diff
@@ -130,4 +133,22 @@ describe('ContentLink', () => {
     expect(event.preventDefault).not.toHaveBeenCalled();
     expect(clickSpy).not.toHaveBeenCalled();
   });
+
+  it('does not call onClick or dispatch if user decides not to discard changes' , () => {
+    const clickSpy = jest.fn();
+    store.dispatch(setAnnotationChangesPending(true));
+    const component = renderer.create(<Provider store={store}>
+      <ConnectedContentLink book={book} page={page} onClick={clickSpy} />
+    </Provider>);
+
+    const event = {
+      preventDefault: jest.fn(),
+    };
+
+    component.root.findByType('a').props.onClick(event);
+
+    expect(dispatch).not.toHaveBeenCalledWith(push(expect.anything()));
+    expect(event.preventDefault).toHaveBeenCalled();
+    expect(clickSpy).not.toHaveBeenCalled();
```

Comment:
these aren't testing anything. `onClick` is an async function and you're not waiting for it to complete, so the functions wouldn't be called either way.

you should make a corresponding test where the confirmation is displayed but you do confirm it, and you'll see the issue.